big_tests/tests/mam_SUITE.erl

```diff
@@ -1261,7 +1261,7 @@ simple_text_search_request(Config) ->
         escalus:send(Alice, escalus_stanza:chat_to(Bob, <<"Also my bike broke down so I'm unable ",
                                                           "to return him home">>)),
         escalus:send(Alice, escalus_stanza:chat_to(Bob, <<"Cats are awesome by the way">>)),
-        maybe_wait_for_archive(Config),
```

Comment:
we can even remove `maybe_wait_for_archive` at all, if it's not used.
Including `archive_wait` flag.

Reply:
`maybe_wait_for_archive` is still used. I replaced it with `wait_for_archive_size` in places where it was safe to use (`escalus:fresh_story`) and easy to know how many messages were put to the archive without much digging in the code.
I agree that replacing `maybe_wait_for_archive` with sth more stable and efficient is a good idea. This is better do to in a separate PR though.